### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by kulibali**
Adds a Windows build script `make.ps1` and updates the code and tests to work on Windows.

### CODE REVIEW

**Comment by ponylang-main** [comment]
Hi @kulibali,

The changelog - added label was added to this pull request; all PRs with a changelog label need to have release notes included as part of the PR. If you haven't added release notes already, please do.

Release notes are added by creating a uniquely named file in the `.release-notes` directory. We suggest you call the file `67.md` to match the number of this pull request.

The basic format of the release notes (using markdown) should be:

```
## Title

End user description of changes, why it's important,
problems it solves etc.

If a breaking change, make sure to include 1 or more
examples what code would look like prior to this change
and how to update it to work after this change.
```

Thanks.

**Comment by kulibali** [comment]
This will need to wait for https://github.com/ponylang/ponyc/pull/3817 and https://github.com/ponylang/ponyc/pull/3816

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@
 TAGS
 tags
 .coverage/
+.vscode
+lock.json
```
Comment (SeanTAllen) [inline .gitignore:8]: should be a newline here

Comment (kulibali) [reply .gitignore]: Fixed, thanks

**Comment by SeanTAllen** [comment]
@kulibali I pushed an update to run on latest ponyc  as well to get recent tcp changes, but nightly wont have all the changes until tomorrow so we should do a fake change tomorrow to get this to run then as well.

**Comment by SeanTAllen** [comment]
@kulibali can you undo the "kick ci commit" and then this can be merged?

**Comment by kulibali** [comment]
@SeanTAllen should be good to merge now.

**Pull request merged**